test/unit/test_qpu_client.py

```diff
@@ -49,240 +48,213 @@ def test_init__sets_processor_and_timeout(mock_engagement_manager: mock.MagicMoc
-                "dtype": "float64",
-                "data": b"buffer-data",
-            },
+    assert qpu_client.get_buffers(request) == GetBuffersResponse(
+        buffers={
+            "ro": BufferResponse(
+                shape=(1000, 2),
+                dtype="float64",
+                data=b"buffer-data",
+            )
         }
-
-    with run_rpcq_server(rpcq_server_with_auth, port):
-        request = GetBuffersRequest(
-            job_id="some-job",
-            wait=True,
-        )
-        assert qpu_client.get_buffers(request) == GetBuffersResponse(
-            buffers={
-                "ro": BufferResponse(
-                    shape=(1000, 2),
-                    dtype="float64",
-                    data=b"buffer-data",
-                )
-            }
-        )
+    )
+    client.call.assert_called_once_with(
+        "get_buffers",
+        job_id=job_id,
+        wait=wait,
+    )
 
 
 def test_fetches_engagement_for_quantum_processor_on_request(
     mock_engagement_manager: mock.MagicMock,
     engagement_credentials: EngagementCredentials,
-    rpcq_server_with_auth: rpcq.Server,
-    port: int,
+    mocker: MockerFixture,
 ):
+    processor_id = "some-processor"
     qpu_client = QPUClient(
-        quantum_processor_id="some-processor",
+        quantum_processor_id=processor_id,
         engagement_manager=mock_engagement_manager,
         request_timeout=3.14,
     )
-
-    def mock_get_engagement(quantum_processor_id: str, request_timeout: float) -> EngagementWithCredentials:
-        assert quantum_processor_id == "some-processor"
-        assert request_timeout == qpu_client.timeout
-        return engagement(
-            quantum_processor_id="some-processor",
+    mock_engagement_manager.get_engagement.return_value = engagement(
+            quantum_processor_id=processor_id,
             seconds_left=9999,
             credentials=engagement_credentials,
-            port=port,
+            port=1234,
         )
 
-    mock_engagement_manager.get_engagement.side_effect = mock_get_engagement
-
-    @rpcq_server_with_auth.rpc_handler
-    def execute_qpu_request(request: rpcq.messages.QPURequest, user: str, priority: int):
-        return ""
+    patch_rpc_client(mocker=mocker, return_value="")
 
-    with run_rpcq_server(rpcq_server_with_auth, port):
-        request = RunProgramRequest(
-            id="",
-            priority=0,
-            program="",
-            patch_values={},
-        )
-        qpu_client.run_program(request)
+    request = RunProgramRequest(
+        id="",
+        priority=0,
+        program="",
+        patch_values={},
+    )
+    qpu_client.run_program(request)
+    mock_engagement_manager.get_engagement.assert_called_once_with(
+        quantum_processor_id=processor_id,
+        request_timeout=qpu_client.timeout,
+    )
 
 
-def test_sets_timeout_on_requests__engagement_expires_later(
+def test__calculate_timeout_engagement_is_longer_than_timeout(
```

Comment:
nit: could you put an extra underscore before `engagement _is_longer...` for readability?